adapters/handlers/rest/configure_api.go

```diff
@@ -377,42 +377,88 @@ func MakeAppState(ctx context.Context, options *swag.CommandLineOptionsGroup) *s
 		migrator.RecountProperties(ctx)
 	}
 
-	// TODO_RAFT START
+	// TODO-RAFT START
 	appState.MetaStore = &fsm
 
-	boostrapper := false
+	// If bootstrap expect is bigger than 0, we are bootstraping a new cluster
+	bootstrapper := appState.ServerConfig.Config.Raft.BootstrapExpect > 0
+
+	if bootstrapper {
+		// The bootstrap process follows these steps:
+		// 1. Ensure all the nodes we expect are running and joined in memberlist
+		// 2. Ensure we can resolve an address for the nodes in raft join
+		// 3. Build a candidate list from the address in memberlist and the port listed in join parameter
+		// 4. Open the FSM store and bootstrap the raft cluster
+
+		// Add nodes from bootstrap join parameter to the raft candidate list
+		candidateList := make([]schemav2.Candidate, 0, len(appState.ServerConfig.Config.Raft.Join))
+		for _, joinNodeNameAndPort := range appState.ServerConfig.Config.Raft.Join {
+			joinNodeNameAndPortSplitted := strings.Split(joinNodeNameAndPort, ":")
+			// shorter names for clearer usage below
+			joinNodeName := joinNodeNameAndPortSplitted[0]
+			joinNodePort := joinNodeNameAndPortSplitted[1]
+
+			waitForNodeToBeInMemberList := func(nodeName string) string {
```

Comment:
I would try to factor `waitForNodeToBeInMemberList := func(nodeName string) string {` out of the loop, for better readabilty.